core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -1032,7 +1063,15 @@ public MetaData build() {
                 }
             }
             aliasAndIndexLookup = Collections.unmodifiableSortedMap(aliasAndIndexLookup);
-            return new MetaData(clusterUUID, version, transientSettings, persistentSettings, indices.build(), templates.build(), customs.build(), allIndices, allOpenIndices, allClosedIndices, aliasAndIndexLookup);
+            if (graveyardBuilder != null) {
+                // if the index graveyard has been modified, add the new version to the custom map
+                customs.put(IndexGraveyard.TYPE, graveyardBuilder.build());
+            } else if (customs.get(IndexGraveyard.TYPE) == null) {
+                // it was never part of the customs map, so add a fresh entry
+                customs.put(IndexGraveyard.TYPE, IndexGraveyard.builder().build());
+            }
```

Comment:
Do we need a graveyard builder here? can't we leave it as a custom  (+ sugar access methods) and people can build it out of class? the only thing we need to do is make sure it's non null (by creating an empty one) when we construct either the build or the MetaData